### PR TITLE
fix(header/p2p): fix crash in server

### LINKED ISSUE #1564: Segmentation fault after `requested headers not found`
### Celestia Node version

v0.6.2-rc1

### OS

Linux, FreeBSD

### Install tools

_No response_

### Others

_No response_

### Steps to reproduce it

```
./build/celestia full start --core.ip https://limani.celestia-devops.dev --p2p.network arabica
```

### Expected result

Node syncs fine.

### Actual result

```
2023-01-05T11:34:29.832+0100    WARN    header/p2p      p2p/server.go:186       server: requested headers not found     {"from": 14402, "to": 14466}
panic: runtime error: invalid memory address or nil pointer dereference 
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x1cd67ba]

goroutine 3317 [running]:
github.com/celestiaorg/celestia-node/header/p2p.(*ExchangeServer).requestHandler(0xc007a3efc0, {0x31c2690, 0xc00d9f8800})
        /home/tomek/opt/celestia-node/header/p2p/server.go:138 +0x67a
github.com/libp2p/go-libp2p/p2p/host/basic.(*BasicHost).SetStreamHandler.func1({0xc00d9ccfc0, 0x1b}, {0x7f3f6c86d7f8?, 0xc00d9f8800})
        /home/tomek/go/pkg/mod/github.com/libp2p/go-libp2p@v0.21.0/p2p/host/basic/basic_host.go:568 +0x76
created by github.com/libp2p/go-libp2p/p2p/host/basic.(*BasicHost).newStreamHandler
        /home/tomek/go/pkg/mod/github.com/libp2p/go-libp2p@v0.21.0/p2p/host/basic/basic_host.go:411 +0x715
```

### Relevant log output

_No response_

### Notes

This happened multiple times, on multiple machines.

### PR DESCRIPTION
**Pull request by vgonkivs**
## Overview
Resolves #1564 + small fixes in the peerTracker
## Checklist
- [x] Required CI checks are passing
- [x] Visual proof for any user facing features like CLI or documentation updates
- [x] Linked issues closed with keywords

### CODE REVIEW

File: header/p2p/exchange.go [hunk 1]
```diff
@@ -75,13 +75,13 @@ func (ex *Exchange) Start(context.Context) error {
...
 
 	go ex.peerTracker.gc()
-	go ex.peerTracker.track()
 	for _, p := range ex.trustedPeers {
 		// Try to pre-connect to trusted peers.
 		// We don't really care if we succeed at this point
 		// and just need any peers in the peerTracker asap
 		go ex.host.Connect(ex.ctx, peer.AddrInfo{ID: p}) //nolint:errcheck
 	}
+	go ex.peerTracker.track()
```
Comment (Wondertan) [inline header/p2p/exchange.go:84]: Does the order matters?

File: header/p2p/exchange_test.go [hunk 2]
```diff
@@ -311,9 +311,11 @@ func createP2PExAndServer(t *testing.T, host, tpeer libhost.Host) (*Exchange, *h
 	require.NoError(t, err)
 	ex, err := NewExchange(host, []peer.ID{tpeer.ID()}, "private", connGater)
 	require.NoError(t, err)
-	ex.peerTracker.trackedPeers[tpeer.ID()] = &peerStat{peerID: tpeer.ID(), peerScore: 100}
 	require.NoError(t, ex.Start(context.Background()))
-
+	time.Sleep(time.Millisecond * 100) // give peerTracker time to add a trusted peer
+	ex.peerTracker.peerLk.Lock()
+	ex.peerTracker.trackedPeers[tpeer.ID()] = &peerStat{peerID: tpeer.ID(), peerScore: 100.0}
```
Comment (Wondertan) [inline header/p2p/exchange_test.go:317]: Why do we need to wait if we add it manually?

Comment (vgonkivs) [reply header/p2p/exchange_test.go:317]: tpeer.ID is a trusted peer that is added on peerTracker starts. Here I re-insert it to the map.

File: header/p2p/exchange.go (hunk 1 above)
Comment (Wondertan) [reply header/p2p/exchange.go:84]: It's any way up to the scheduler to decide what goroutine will be picked first. 

Just grouping wise, pls keep track and gc `go`s together.

**Review by Wondertan — APPROVED** [review]

**Pull request merged**